tests/app/questionnaire/test_path_finder.py

```diff
@@ -653,6 +653,124 @@ def test_repeating_groups_conditional_location_path(self):
+            sut_path = sut.get_full_routing_path()
+
+            # validate that path returned from path finder matches expected path
+
+            self.assertEqual(expected_path, sut_path)
+
+    def test_repeating_group_block_skip_correct_if_independent_answer_is_not_repeating_but_does_not_cause_a_skip(self):
+        """ tests blocks are correctly added to path in the case where the blocks in a repeating group have
+        skip conditions and those skip conditions are dependant on a non repeating answer which causes a skip
+        condition to return false.
+        """
+
+        independent_answer = 2
+
+        with patch('app.questionnaire.rules._answer_is_in_repeating_group', return_value=False):
+            expected_path = [
+                Location('about-household-group', 0, 'household-composition'),
+                Location('independent-answer-group', 0, 'non-repeating-question-block'),
+                Location('dependant-group', 0, 'additional-question-block'),
+                Location('dependant-group', 1, 'additional-question-block'),
+                Location('dependant-group', 2, 'additional-question-block'),
+                Location('questionnaire-completed', 0, 'confirmation')
+            ]
+
+            schema = load_schema_from_params('test',
+                                             'skip_conditions_from_repeating_group_based_on_non_repeating_answer')
+            answer_store = AnswerStore()
+
+            # Set up some first names
+            answer_store.add(Answer(answer_id='first-name', value='aaa', group_instance=0))
+            answer_store.add(Answer(answer_id='first-name', value='bbb', group_instance=1))
+            answer_store.add(Answer(answer_id='first-name', value='ccc', group_instance=2))
+
+            # Set up the independent answer
+            answer_store.add(Answer(answer_id='an-independent-answer', value=independent_answer, group_instance=0))
+
+            sut = PathFinder(schema, answer_store=answer_store, metadata={}, completed_blocks=[])
+            sut_path = sut.get_full_routing_path()
+
+            # validate that path returned from path finder matches expected path
+
+            self.assertEqual(expected_path, sut_path)
+
+    def test_repeating_group_block_skip_correct_if_independent_answer_is_repeating(self):
+        """ tests blocks are correctly added to path in the case where the blocks have
+        skip conditions and those skip conditions are dependant on a repeating answer
+        tests various combinations of independent answers being set , since we saw different
+        behaviours for group_instance 0 as it was the default"""
+
+        param_list = [(0, 2), (1, 3)]
+
+        with patch('app.questionnaire.rules._answer_is_in_repeating_group', return_value=True):
+            for first_group_instance, second_group_instance in param_list:
+
+                with self.subTest():
```

Comment:
Could you pass the context to sub test so it's included in any test failure message?

```
with self.subTest(first_group_instance=first_group_instance, second_group_instance=second_group_instance):
```

Reply:
Not sure we're talking about the same thing? The subTest() method accepts arguments purely for the purpose of displaying what iteration is currently being tested in case of a failure.

Reply:
No I was talking about almost duplicate tests , I see what you mean , will add message